### DISCUSSION TITLE
Remap signal generator pin

### DISCUSSION THREAD

**Discussion by marcojk** [fhdm-dev/scoppy#79] [General]
Hi and thanks for this great project.

I'm using one of the many rpi2040 board around, specifucally Seeed XIAO RP2040. Tiny form factor, working perfectly.
Unfortunately it doesn't expose pin 22 for the signal generator output.
I've skimmed through the firmware code and it looks like it can, of course, attach the generated signal to any pin.
But this option is missing in the phone app.
Thanks a lot.

**Comment by fhdm-dev** [comment]
Hi,

Yes. That's correct. Currently the only way to change the pin used for the signal generator is to re-compile the firmware.

I must admit I haven't really put any thought into how the firmware could work with non-pico boards. It probably wouldn't be too difficult to make most of the pin mappings configurable via the app. I'll definitely add that to my to-do list :smile:

**Comment by fhdm-dev** [comment]
The GPIO mappings can now be configured via the app (the latest version of app and firmware are required). Tap the connection badge at the lower left of the screen (the one that shows 'USB OK') and then tap 'Connected device' and 'Firmware settings'.

**Reply by marcojk** [reply]
Great news! Thanks, a very needed feature
